support/k8s/k8s-workload-registrar/controller_test.go

```diff
@@ -387,6 +421,72 @@ func TestControllerAnnotationBasedRegistration(t *testing.T) {
 	}, r.GetEntries())
 }
 
+func TestControllerFederationBasedRegistration(t *testing.T) {
+	controller, r := newTestController("", "")
+
+	// Send in a POD CREATE and assert that it will be admitted
+	requireReviewAdmissionSuccess(t, controller, &admv1beta1.AdmissionRequest{
+		UID: "uid",
+		Kind: metav1.GroupVersionKind{
+			Version: "v1",
+			Kind:    "Pod",
+		},
+		Namespace: "NAMESPACE",
+		Name:      "PODNAME",
+		Operation: "CREATE",
+		Object: runtime.RawExtension{
+			Raw: []byte(fakePodWithFederation),
+		},
+	})
+
+	// Assert that the registration entry for the pod was created
+	requireEntriesEqual(t, []*types.Entry{
```

Comment:
Something here is causing infinite recursion and causing the test to run out of memory...